src/main/scala/scorex/core/network/NodeViewSynchronizer.scala

```diff
@@ -121,13 +147,15 @@ class NodeViewSynchronizer[TX <: Transaction,
 
   protected def getLocalSyncInfo: Receive = {
     case SendLocalSyncInfo =>
-      historyReaderOpt.foreach(r => sendSync(statusTracker, r))
+      historyReaderOpt.foreach(r => sendSync(statusTracker, r.syncInfo))
   }
 
-  protected def sendSync(syncTracker: SyncTracker, history: HR): Unit = {
+  protected def sendSync(syncTracker: SyncTracker, syncInfo: SI, sendToRandomIfEmpty: Boolean = false): Unit = {
```

Comment:
This new behavior is buggy, as if "sendToRandomIfEmpty" =true, "updateLastSyncSentTime" is not to be called on the random peer in SyncTracker. In general, new the logic to choose a peer is cluttered with no good reason.

Reply:
From performance point of view we need a way to send `Sync` message at least to someone, when we have already received and applied all modifiers, but are not synced with other peers yet.
Removed this change for now as far as it is not crucial, may be done in a separate PR.